### ISSUE TITLE
terraform should exit gracefully after error locking state

### ISSUE THREAD

**Issue by vchokshi** [hashicorp/terraform#20000]
Terraform v0.11.11
+ provider.aws v1.55.0

Problem: When using Terraform with Dynamo to have a shared lock file, if we create contention the HTTP 400 response to the second terraform apply command causes TF to throw an error and exit code 1. 

Enhancement Suggestion: The user should receive a WARN that TF was unable to continue due to an existing lock, and TF should should exit gracefully.

Steps to reproduce: 
1. Setup a shared lock with AWS Dynamo
2. in one shell, terraform apply
3. in a second shell, terraform apply.

See the error in the second shell

```
Error: Error locking state: Error acquiring the state lock: ConditionalCheckFailedException: The conditional request failed
	status code: 400, request id: L7JC2DIC4O79L0JGC9SC4Q2NCRVV4KQNSO5AEMVJF66Q9ASUAAJG
Lock Info:
  ID:        ad45f03d-8b49-35d0-0128-6f1a403cfd7f
  Path:      <omitted>/terraform.tfstate
  Operation: OperationTypeApply
  Who:       vchokshi@iot4.net
  Version:   0.11.11
  Created:   2019-01-14 19:30:34.548419 +0000 UTC
  Info:
```

I'd be happy to look into fixing this if the community agrees and someone can help me get started! 

Vihar

### Terraform Version

```
...
```

### Terraform Configuration Files

```hcl
...
```

**Comment by mildwonkey** [comment]
Hi @vchokshi !
Thank you for submitting this feature request. I updated the issue title to focus on the exit code after a state locking error.

Can you please share your reasoning for this feature request? My thoughts are that, if a user is running terraform in automation, this scenario should produce an error to indicate a failed run. I'd like to hear your use-case.

**Comment by vchokshi** [comment]
HI @mildwonkey !

Much of my reasoning comes from this [link](https://airbrake.io/blog/http-errors/400-bad-request), which I uncovered _after_ setting up the Dynamo lock for the first time. I was testing the contention with a peer and when I saw the error thrown; I thought to myself 'oh snap it broke', but then upon further review I realized the locking mechanism worked as expected, but as the user, I was briefly confused. 

Out TF code will be handed off to others to run; as the expert TF guy in the org I figured if I was confused, so might others.

Given the output in the subsequent error message that listed out the details of the lock, I realized Terraform was refusing to execute, not failing to execute. 

The author of the above [link](https://airbrake.io/blog/http-errors/400-bad-request) went into great detail on the HTTP 400 code, and I felt their post was definitive. 

Does this make sense?
